src/Api/FacetApi.php

```diff
@@ -63,6 +94,20 @@ protected function getRequestMultipartStreamElements(): array
         return $parts;
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function setSort($sort): SearchApiBaseInterface
```

Comment:
It seems that facets sort is a different kind of fish. For example:
- Search sort can be null, Facet sort cannot
- Search sort has no default value, Facet does

...and so on. So we shouldn't extend but we need them as different methods.

Also it seems that the search sort needs a model, same as metadata (will open a ticket)